cli/install/install.go

```diff
@@ -0,0 +1,776 @@
+		err := util.CreateSymLink(version, bindst, "tarantool", true)
+		log.Infof("Done")
+		return err
+	} else if checkExisting(version, bindst) && flags.Reinstall {
+		log.Infof("%s version of tarantool already exists, removing binary...", version)
+		os.RemoveAll(filepath.Join(bindst, version))
+	}
+
+	if checkExisting(version, headerDir) && !flags.Reinstall {
+		log.Infof("%s version of tarantool headers already exists, updating symlink...", version)
+		err := util.CreateSymLink(version, headerDir, "tarantool", true)
+		log.Infof("Done")
+		return err
+	} else if checkExisting(version, bindst) && flags.Reinstall {
+		log.Infof("%s version of tarantool binary already exists, removing headers...", version)
+		os.RemoveAll(filepath.Join(headerDir, version))
+	}
+	return err
+}
+
+// installTarantool installs selected version of tarantool.
+func installTarantool(version string, bindst string, headerDir string, flags InstallationFlag,
+	distfiles string) error {
+	tarVersion := ""
+
+	// Get latest tarantool tag if needed.
+	if version == "tarantool" {
+		tarVersion, _ = getLatestTarantoolTag()
+		version += "=" + tarVersion
+	} else {
+		tarVersion = strings.TrimPrefix(version, "tarantool=")
+	}
+	if util.IsDeprecated(tarVersion) {
+		return fmt.Errorf("Deprecated version of tarantool")
+	}
+
+	// Check bin and header dirs.
+	if bindst == "" {
+		return fmt.Errorf("Cannot procces bindir.")
+	}
+	if headerDir == "" {
+		return fmt.Errorf("Cannot procces headerdir.")
+	}
+	logFile, _ := ioutil.TempFile("", "tarantool_install")
+	defer os.Remove(logFile.Name())
+
+	log.Infof("Installing tarantool=" + tarVersion)
+
+	// Check if program is already installed.
+	log.Infof("Checking existing...")
+	err := checkExistingTarantool(version, bindst, headerDir, flags)
+	if err != nil {
+		return err
+	}
+
+	// Check dependencies.
+	log.Infof("Checking dependencies...")
+	if flags.Force {
+		log.Infof("Skipped...")
+	} else if !checkProgrammDependencies("tarantool") {
+		return nil
+	}
+
+	path, err := os.MkdirTemp("", "tarantool_install")
+	os.Chmod(path, 0777)
+	if err != nil {
+		return err
+	}
+	if !flags.Noclean {
+		defer os.RemoveAll(path)
+	}
+
+	// Download tarantool.
+	if flags.Local {
+		log.Infof("Checking local files...")
+		if checkExisting("tarantool", distfiles) {
+			log.Infof("Local files found, installing from them...")
+			localPath, _ := util.JoinAbspath(distfiles, "tarantool")
+			err = util.CopyDir(localPath, path)
+			if err != nil {
+				printLog(logFile.Name())
+				return err
+			}
+			util.ExecuteCommand("git", flags.Verbose, logFile, path, "checkout", tarVersion)
+		} else {
+			return fmt.Errorf("Can't find distfiles directory.")
+		}
+	} else {
+		log.Infof("Downloading tarantool...")
+		downloadRepo(TarantoolLink, tarVersion, path, logFile, flags.Verbose)
+		if err != nil {
+			printLog(logFile.Name())
+			return err
+		}
+	}
+
+	// Build tarantool.
+	log.Infof("Building tarantool...")
```

Comment:
Please, don't make such huge functions. 
Download tarantool - separate function. 
Build tarantool in local directory - separate function.
etc.
This functionality can be easily covered with unit tests.

Reply:
Separated